apps/els_lsp/src/els_code_action_provider.erl

```diff
@@ -31,8 +31,12 @@ handle_request({document_codeaction, Params}, State) ->
 
 %% @doc Result: `(Command | CodeAction)[] | null'
 -spec code_actions(uri(), range(), code_action_context()) -> [map()].
-code_actions(Uri, _Range, #{<<"diagnostics">> := Diagnostics}) ->
-  lists:flatten([make_code_action(Uri, D) || D <- Diagnostics]).
+code_actions(Uri, Range, #{<<"diagnostics">> := Diagnostics}) ->
+  Actions = lists:flatten([make_code_action(Uri, D) || D <- Diagnostics]),
+  case application:get_env(els_core, wrangler_enabled) of
```

Comment:
Ideally, this module wouldn't know about Wrangler or other specific "backends".
The assumption here was that code actions are linked to diagnostics, which is not true in your case.
I wonder if we should make this a bit more general and allow external code action providers, not necessarily linked to diagnostics.

Reply:
Yes, you are right. I can attend the meeting on Friday to discuss.